crates/bevy_solari/src/scene/blas.rs

```diff
@@ -0,0 +1,133 @@
+    {
+        blas_manager.remove(asset_id);
+    }
+
+    if extracted_meshes.extracted.is_empty() {
+        return;
+    }
+
+    // Create new BLAS for added or changed meshes
+    let blas_resources = extracted_meshes
+        .extracted
+        .iter()
+        .filter(|(_, mesh)| is_mesh_raytracing_compatible(mesh))
+        .map(|(asset_id, _)| {
+            let vertex_slice = mesh_allocator.mesh_vertex_slice(asset_id).unwrap();
+            let index_slice = mesh_allocator.mesh_index_slice(asset_id).unwrap();
+
+            let (blas, blas_size) =
+                allocate_blas(&vertex_slice, &index_slice, asset_id, &render_device);
+
+            blas_manager.insert(*asset_id, blas);
+
+            (*asset_id, vertex_slice, index_slice, blas_size)
+        })
+        .collect::<Vec<_>>();
+
+    // Build geometry into each BLAS
+    let build_entries = blas_resources
+        .iter()
+        .map(|(asset_id, vertex_slice, index_slice, blas_size)| {
+            let geometry = BlasTriangleGeometry {
+                size: blas_size,
+                vertex_buffer: vertex_slice.buffer,
+                first_vertex: vertex_slice.range.start,
+                vertex_stride: 48,
+                index_buffer: Some(index_slice.buffer),
+                first_index: Some(index_slice.range.start),
+                transform_buffer: None,
+                transform_buffer_offset: None,
+            };
+            BlasBuildEntry {
+                blas: &blas_manager[asset_id],
+                geometry: BlasGeometries::TriangleGeometries(vec![geometry]),
+            }
+        })
+        .collect::<Vec<_>>();
+
+    let mut command_encoder = render_device.create_command_encoder(&CommandEncoderDescriptor {
+        label: Some("build_blas_command_encoder"),
+    });
+    command_encoder.build_acceleration_structures(&build_entries, &[]);
+    render_queue.submit([command_encoder.finish()]);
+}
+
+fn allocate_blas(
+    vertex_slice: &MeshBufferSlice,
+    index_slice: &MeshBufferSlice,
+    asset_id: &AssetId<Mesh>,
+    render_device: &RenderDevice,
+) -> (Blas, BlasTriangleGeometrySizeDescriptor) {
+    let blas_size = BlasTriangleGeometrySizeDescriptor {
+        vertex_format: Mesh::ATTRIBUTE_POSITION.format,
+        vertex_count: vertex_slice.range.len() as u32,
+        index_format: Some(IndexFormat::Uint32),
+        index_count: Some(index_slice.range.len() as u32),
+        flags: AccelerationStructureGeometryFlags::OPAQUE,
+    };
+
+    let blas = render_device.wgpu_device().create_blas(
+        &CreateBlasDescriptor {
+            label: Some(&asset_id.to_string()),
+            flags: AccelerationStructureFlags::PREFER_FAST_TRACE,
+            update_mode: AccelerationStructureUpdateMode::Build,
+        },
+        BlasGeometrySizeDescriptors::Triangles {
+            descriptors: vec![blas_size.clone()],
+        },
+    );
+
+    (blas, blas_size)
+}
+
+fn is_mesh_raytracing_compatible(mesh: &Mesh) -> bool {
```

Comment:
Should we log an error in this function if the mesh isn't compatible? It might help track down why those scenes weren't working when I tested them.

Reply:
Yeah this was brought up above, I can add it to the next PR.